### ISSUE TITLE
Filter with subquery throws FormatException

### ISSUE THREAD

**Issue by EndlessDelirium** [f3ath/jessie#43]
I'm unable to run JsonPaths that contain a filter with a subquery similar to [example.dart Line 79](https://github.com/f3ath/jessie/blob/b499b1cba66360121ecc988abafe66338ccce50d/example/example.dart#L79):
```
JsonPath(r'$.store.book[?@.category == $.store.book[-1].category].title')
```

When running this example against v0.4.4 it fails with the same FormatException as it does in my own code

```
FormatException (FormatException: end of input expected at 1:13. Expression: $.store.book[?@.category == $.store.book[-1].category].title)
```

Any idea what's wrong?

Btw, thanks a lot for this truly awesome library!

**Comment by EndlessDelirium** [comment]
That would be great! Thanks a lot!

**Comment by EndlessDelirium** [comment]
Now works flawlessly for our use cases in the just released 0.5.0-beta1. Awesome!
